### PR TITLE
[Merged by Bors] - fix: add `-lLake` argument to `pole` and `mk_all`

### PR DESCRIPTION
**Pull request by grunweg**
Since Lean 4.9, the behaviour of `supportInterpreter` executables has changed, making this necessary:
both of these executables require modules from `lake`.
In contrast, `shake` also uses `supportInterpreter`, but does not depend on `lake`.
[zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/automatically.20generate.20ProjectName.2Elean/near/444239597)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
[On zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/automatically.20generate.20ProjectName.2Elean), @Rida-Hamadani reports successfully testing this on windows.

**Comment by Rida-Hamadani** [comment]
It succeeds only after deleting `.lake`, otherwise it gives an error.

File: lakefile.lean [hunk 1]
```diff
@@ -62,6 +62,7 @@ lean_exe checkYaml where
 lean_exe mk_all where
   srcDir := "scripts"
   supportInterpreter := true
+  weakLinkArgs := #["-lLake"]
```
Comment (semorrison) [inline lakefile.lean:65]:
```suggestion
  -- Executables which import `Lake` must set `-lLake`.
  weakLinkArgs := #["-lLake"]
```

**Comment by semorrison** [comment]
bors d+

File: lakefile.lean [hunk 2]
```diff
@@ -76,6 +77,7 @@ and then calculates the longest pole
 lean_exe pole where
   root := `LongestPole.Main
   supportInterpreter := true
+  weakLinkArgs := #["-lLake"]
```
Comment (grunweg) [inline lakefile.lean:80]:
```suggestion
  -- Executables which import `Lake` must set `-lLake`.
  weakLinkArgs := #["-lLake"]

File: lakefile.lean (hunk 1 above)
Comment (grunweg) [reply lakefile.lean:65]: Good idea, thanks - added.

**Comment by grunweg** [comment]
Thanks for the review!
bors r+

**Pull request closed without merging**